Lib/multiprocessing/pool.py

```diff
@@ -189,12 +210,23 @@ def __init__(self, processes=None, initializer=None, initargs=(),
                 p.join()
             raise
 
+        if hasattr(self._outqueue, "_reader"):
+            task_queue_sentinels = [self._outqueue._reader]
+        else:
+            task_queue_sentinels = []
+        if hasattr(self._change_notifier, "_reader"):
+            self_notifier_sentinels = [self._change_notifier._reader]
+        else:
+            self_notifier_sentinels = []
+
+        sentinels = [*task_queue_sentinels, *self_notifier_sentinels]
```

Comment:
How about a dedicated method that returns this list and that you can override in `ThreadPool`?